src/test/java/org/folio/rest/api/HridSettingsStorageTest.java

```diff
@@ -423,38 +509,82 @@ public void canRollbackFailedTransaction(TestContext testContext) {
-                  assertThat(currentHridSettings.getInstances().getPrefix(),
-                      is(originalHridSettings.getInstances().getPrefix()));
-                  assertThat(currentHridSettings.getInstances().getStartNumber(),
-                      is(originalHridSettings.getInstances().getStartNumber()));
-                  assertThat(currentHridSettings.getHoldings().getPrefix(),
-                      is(originalHridSettings.getHoldings().getPrefix()));
-                  assertThat(currentHridSettings.getHoldings().getStartNumber(),
-                      is(originalHridSettings.getHoldings().getStartNumber()));
-                  assertThat(currentHridSettings.getItems().getPrefix(),
-                      is(originalHridSettings.getItems().getPrefix()));
-                  assertThat(currentHridSettings.getItems().getStartNumber(),
-                      is(originalHridSettings.getItems().getStartNumber()));
-                  return Future.succeededFuture(currentHridSettings);
-                })
-                .onComplete(promise);
-          });
-          return promise.future();
-        })
-        .onComplete(testContext.asyncAssertSuccess(
-            v1 -> log.info("Finished canRollbackFailedTransaction()")));
+      .compose(originalHridSettings -> {
+        Promise<HridSettings> promise = Promise.promise();
+        hridManager.updateHridSettings(newHridSettings).onComplete(ar -> {
+          assertTrue(ar.failed());
+          hridManager.getHridSettings()
+            .compose(currentHridSettings -> {
+              assertThat(currentHridSettings.getId(), is(originalHridSettings.getId()));
+              assertThat(currentHridSettings.getInstances().getPrefix(),
+                is(originalHridSettings.getInstances().getPrefix()));
+              assertThat(currentHridSettings.getInstances().getStartNumber(),
+                is(originalHridSettings.getInstances().getStartNumber()));
+              assertThat(currentHridSettings.getHoldings().getPrefix(),
+                is(originalHridSettings.getHoldings().getPrefix()));
+              assertThat(currentHridSettings.getHoldings().getStartNumber(),
+                is(originalHridSettings.getHoldings().getStartNumber()));
+              assertThat(currentHridSettings.getItems().getPrefix(),
+                is(originalHridSettings.getItems().getPrefix()));
+              assertThat(currentHridSettings.getItems().getStartNumber(),
+                is(originalHridSettings.getItems().getStartNumber()));
+              return Future.succeededFuture(currentHridSettings);
+            })
+            .onComplete(promise);
+        });
+        return promise.future();
+      })
+      .onComplete(testContext.asyncAssertSuccess(
+        v1 -> log.info("Finished canRollbackFailedTransaction()")));
+  }
+
+  @Test
+  public void canRollbackFailedTransactionWithoutLeadingZeroes(TestContext testContext) {
```

Comment:
Why is this test needed? Why does not having leading zeros make it significantly different to `canRollbackFailedTransaction` above it?

Reply:
Hmm... I wrote these tests during investigation how HridManager works. These tests hasn't big sense.

Reply:
> These tests hasn't big sense.

Are you referring to the new tests?

Reply:
I mean that I have to delete this two tests because these tests not needed to testing my changes. New changes do not affect on Long Type overflow and on transaction rollback. I wrote them for investigating how HridManager works.

Reply:
Thanks for explaining that. Does that mean those new tests will be removed?

Reply:
Yes. I will remove them.

Reply:
Fixed.